kubernetes/resource_kubernetes_job_v1.go

```diff
@@ -48,6 +51,78 @@ func resourceKubernetesJobV1() *schema.Resource {
 	}
 }
 
+func resourceKubernetesJobV1CustomizeDiff(ctx context.Context, d *schema.ResourceDiff, meta interface{}) error {
+	if d.Id() == "" {
+		log.Printf("[DEBUG] Resource ID is empty, resource not created yet.")
+		return nil
+	}
+
+	// Retrieve old and new TTL values as strings
+	oldTTLRaw, newTTLRaw := d.GetChange("spec.0.ttl_seconds_after_finished")
+
+	var oldTTLStr, newTTLStr string
+
+	if oldTTLRaw != nil {
+		oldTTLStr, _ = oldTTLRaw.(string)
+	}
+	if newTTLRaw != nil {
+		newTTLStr, _ = newTTLRaw.(string)
+	}
+
+	oldTTLInt, err := strconv.Atoi(oldTTLStr)
+	if err != nil {
+		oldTTLInt = 0
```

Comment:
I'm not sure this makes sense – if the number supplied in the config doesn't even parse as an integer we should produce an error rather than defaulting it to `0` I think.